rest-api-spec/src/main/resources/rest-api-spec/api/fields_enum.json

```diff
@@ -0,0 +1,35 @@
+{
+  "fields_enum":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields-enum.html",
+      "description": "The fields enum API  can be used to discover field names in the index that begin with the provided string. It is designed for low-latency look-ups used in auto-complete scenarios."
+    },
+    "stability":"beta",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "application/json"],
+      "content_type": ["application/json"]
+    },
+    "url":{
+      "paths":[
```

Comment:
Since the path can be either `/_fields_enum` or `/{index}/_fields_enum` we should have multiple entries in `paths`. Here's the [`search` API](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/search.json#L14) as an example of how to do this.